### PR TITLE
add files to bower ignore

### PR DESCRIPTION
**Pull request by timse**
this removes clutter when installing `js-cookie` as a bower component

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -3,5 +3,10 @@
   "main": [
     "src/js.cookie.js"
   ],
-  "ignore": []
+  "ignore": [
+    "travis.sh",
+    "test",
```
Comment (FagnerMartinsBrack) [inline bower.json]: We need this folder to be able to [integrate the encoding tests for server-side frameworks](https://github.com/js-cookie/js-cookie/blob/v2.0.3/CONTRIBUTING.md#integration-with-server-side).

**Comment by timse** [comment]
I'm not sure i understand this. Why would you want to integrate a third parties tests?

File: bower.json (hunk 1 above)
Comment (timse) [inline bower.json]: I'm not sure i understand this. Why would you want to integrate a third parties tests?

**Comment by FagnerMartinsBrack** [comment]
See https://github.com/js-cookie/js-cookie/commit/e95fde0c6205f992dbab7a7d189726cfd2e38213, most of those files are needed in order to run the build after the files are retrieved using bower.

[Here](https://github.com/js-cookie/java-cookie/blob/v0.0.2/Gruntfile.js#L8) is the example of a project that uses it.

> I'm not sure i understand this. Why would you want to integrate a third parties tests?

To prevent having to duplicate each character encoding/decoding test in a different server-side language.

**Comment by FagnerMartinsBrack** [comment]
js-cookie encodes [only the characters that are not allowed in cookies](https://github.com/js-cookie/js-cookie/tree/v2.0.3#encoding), as specified in the RFC 6265. See [here](https://github.com/js-cookie/js-cookie/issues/35) the details. The integration test hook is exposed to be able to ensure consistency between server-side implementations with the default js-cookie decoding/encoding mechanism, like the Java Cookie project.

**Comment by timse** [comment]
> To prevent having to duplicate each character encoding/decoding test in a different server-side language.
integrating the whole javascript toolchain to save up on a test doesnt look too like i want to do that. But i guess you can do whatever you feel like with your package :)

**Comment by FagnerMartinsBrack** [comment]
If you are using bower to hold only production-ready components you are doing something different from what it was built for, see https://github.com/bower/bower/issues/368#issuecomment-15916138

> The intention has never been for Bower to provide production-ready components. If someone wants to register a package that includes minified files to reference in your script/link elements or something, then that's fine. But that's not Bower's concern. Publishing minified files is something that might be more common if we decide to include an npm-like publish model.

**Comment by timse** [comment]
i did read the comment the first time you linked to it. they talk about not providing minified code which i totally agree with, as it would be just more clutter.

they are not talking about adding the whole test suit so you can do weird things with it. 

i'd rather add extract your test case as a bower component and require it in here as a dev dependency but then again its not my package and you are free to do whatever you want with it. I just very much disagree.

**Comment by FagnerMartinsBrack** [comment]
@timse I understand your concerns. One more reason to keep the test folder is to make it possible to run `grunt default` in a custom build environment that consumes js-cookie package, in order to be able to generate the minified file for production.

[We have plans](https://github.com/js-cookie/js-cookie/issues/57) to provide the minified file without having to run the build, mostly for npm. Maybe after that it would make sense moving the encoding tests to a separate package and clear all the clutter for those downloading the full package once and for all.

I will leave this out for @carhartl to decide what's the best way to go for now.

**Comment by carhartl** [comment]
I'm not familiar with how the server-side testing is being setup, but I would assume we could pull in a git tag instead of a bower package to set it up?

If so, I'd prefer not to clutter the bower package, i.e. +1 for this PR.

Regarding the minification it provides at the moment, imo we should not impose Grunt on the users of the package, by now Gulp seems even more popular. That said, the task may be useless too many and declare a dependency that isn't satisfied in the first place. Furthermore, developers likely have set up their own build tool chain, an argument that was often made by us in the past against including a minified version in the git repository. 

If minification etc. is a common expectation for bower packages to provide, then I'll change my mind on that one :)

**Comment by FagnerMartinsBrack** [comment]
> I'm not familiar with how the server-side testing is being setup

It integrates a front-end build with the server-side build routine, executing bower eventually to load js-cookie files.

> I would assume we could pull in a git tag instead of a bower package to set it up?

With [submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules), probably yes.

Very good point about grunt/gulp. Different from [the last removal](https://github.com/carhartl/jquery-cookie/pull/205), now we have a proper justification for removing this.

File: bower.json (hunk 1 above)
Comment (FagnerMartinsBrack) [inline bower.json]: @timse As from [this comment](https://github.com/js-cookie/js-cookie/pull/83#issuecomment-143968146), doesn't it makes sense removing everything then, including the dot files? Like `.gitignore`, `.jshintignore`, `.jshintrc`, `.travis.yml` and `.tm_properties`?

It seems that the only things that matters are `README.md`, `MIT-LICENSE.txt`, `CONTRIBUTING.md` (is it needed?), and `src/js.cookie.js`, right?

**Comment by FagnerMartinsBrack** [comment]
@timse Are you still willing to continue on this?

File: bower.json (hunk 1 above)
Comment (timse) [inline bower.json]: i guess that would be finde with me

Comment (FagnerMartinsBrack) [inline bower.json]: Thanks for the feedback, do you want to update the PR?

Comment (timse) [inline bower.json]: Sure i will as soon as im back at my computer :)

Comment (FagnerMartinsBrack) [inline bower.json]: :+1:

**Comment by FagnerMartinsBrack** [comment]
Thanks man :+1:

**Comment by timse** [comment]
thanks! :) :+1:

**Pull request closed without merging**